draft-ietf-tls-tls13.md

```diff
@@ -1532,55 +1532,68 @@ derivation scheme for all cipher modes.
 The basic TLS Handshake for DH is shown in {{tls-full}}:
 
 ~~~
-     Client                                               Server
-
-     ClientHello
-       + KeyShare              -------->
-                                                     ServerHello
-                                                      + KeyShare
-                                           {EncryptedExtensions}
-                                          {ServerConfiguration*}
-                                                  {Certificate*}
-                                           {CertificateRequest*}
-                                            {CertificateVerify*}
-                               <--------              {Finished}
-     {Certificate*}
-     {CertificateVerify*}
-     {Finished}                -------->
-     [Application Data]        <------->      [Application Data]
-
-            +  Indicates extensions sent in the
-               previously noted message.
-
-            *  Indicates optional or situation-dependent
-               messages that are not always sent.
-
-            {} Indicates messages protected using keys
-               derived from the ephemeral secret.
-
-            [] Indicates messages protected using keys
-               derived from the master secret.
+       Client                                               Server
+  
+Key  ^ ClientHello                                                  
+Exch v  + KeyShare               -------->                           
```

Comment:
The ^ and v without a line joining them are a little hard to follow.  What about:

```
v---- Key exchange
 ClientHello
       + KeyShare              -------->
                                                       ServerHello
                                                      + KeyShare
v---- Server parameters
                                             {EncryptedExtensions}
                                          {ServerConfiguration*}
                                           {CertificateRequest*} 
v---- Authentication
                                                    {Certificate*}
                                            {CertificateVerify*}
                                <--------              {Finished}
     {Certificate*}
     {CertificateVerify*}
     {Finished}                -------->
```

Also, this doesn't allow the server to speak first, is that intentional?